modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ResourceF.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.enrich.common.utils
+
+import cats.Id
+
+import cats.effect.{Resource, Sync}
+
+trait ResourceF[F[_]] {
```

Comment:
This is very cool! 😎

Reply:
Thanks ! Glad that you like that. Dealing with `Id` is a pain !

Reply:
Yeah, I can see `Id` is basically everywhere. Out of curiosity: why we need to support `Id` in enrich? I know about rdb loader/shredder and spark thing, but enrich?

Reply:
Hopefully it's going to be a past very soon.

Current Stream Enrich is built on KCL, without anything from Cats Effect / FS2. Also Beam Enrich isn't that distant past (we've just EOL'ed it few months ago). Not to mention we had Spark Enrich at some point.